docs/develop/host-specific-api-model.md

```diff
@@ -0,0 +1,211 @@
+        console.log (tableCount.value);
+    });
+```
+
+### set()
+
+Setting properties on an object with nested navigation properties can be cumbersome. As an alternative to setting individual properties using navigation paths as described above, you can use the `object.set()` method that is available on objects in the promise-based JavaScript APIs. With this method, you can set multiple properties of an object at once by passing either another object of the same Office.js type or a JavaScript object with properties that are structured like the properties of the object on which the method is called.
+
+The following code sample sets several format properties of a range by calling the `set()` method and passing in a JavaScript object with property names and types that mirror the structure of properties in the `Range` object. This example assumes that there is data in range **B2:E2**.
+
+```js
+Excel.run(function (ctx) {
+    var sheet = ctx.workbook.worksheets.getItem("Sample");
+    var range = sheet.getRange("B2:E2");
+    range.set({
+        format: {
+            fill: {
+                color: '#4472C4'
+            },
+            font: {
+                name: 'Verdana',
+                color: 'white'
+            }
+        }
+    });
+    range.format.autofitColumns();
+
+    return ctx.sync();
+}).catch(function(error) {
+    console.log("Error: " + error);
+    if (error instanceof OfficeExtension.Error) {
+        console.log("Debug info: " + JSON.stringify(error.debugInfo));
+    }
+});
+```
+
+## get&#42;OrNullObject methods
+
+Some `get*` methods throw an exception when the desired object is not present. For example, if you attempt to get an Excel worksheet by specifying a worksheet name that is not in the workbook, the `getItem()` method throws an `ItemNotFound` exception.
+
+Any `get*OrNullObject` method variant lets you check for an object without throwing exceptions. These methods return a null object (not the JavaScript `null`) rather than throwing an exception if the specified item doesn't exist. For example, you can call the `getItemOrNullObject()` method on a collection such as **Worksheets** to attempt to retrieve an item from the collection. The `getItemOrNullObject()` method returns the specified item if it exists; otherwise, it returns a null object. The null object that is returned contains the boolean property `isNullObject` that you can evaluate to determine whether the object exists.
+
+The following code sample attempts to retrieve an Excel worksheet named "Data" by using the `getItemOrNullObject()` method. If the method returns a null object, a new sheet needs to be created before actions can taken on the sheet.
+
+```js
+var dataSheet = context.workbook.worksheets.getItemOrNullObject("Data");
+
+return context.sync()
+  .then(function() {
+    if (dataSheet.isNullObject) {
+        // Create the sheet.
+    }
+
+    dataSheet.position = 1;
+    //...
+  })
+```
+
```

Comment:
Just putting a random comment here: you know what might be nice is a flow chart/graphic at the top of this page for people like 
1. Run 
2. Load
etc

Reply:
We need a good drawing of how proxy objects work in general... I'll add it to the refactoring list.